src/topology/algebra/ordered.lean

```diff
@@ -2198,22 +2198,185 @@ lemma tendsto_of_monotone {ι α : Type*} [preorder ι] [topological_space α]
 if H : bdd_above (range f) then or.inr ⟨_, tendsto_at_top_csupr h_mono H⟩
 else or.inl $ tendsto_at_top_at_top_of_monotone' h_mono H
 
-@[to_additive] lemma tendsto_inv_nhds_within_Ioi {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Ioi [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within a (Ioi a)) (nhds_within (a⁻¹) (Iio (a⁻¹))) :=
 (continuous_inv.tendsto a).inf $ by simp [tendsto_principal_principal]
 
-@[to_additive] lemma tendsto_inv_nhds_within_Iio {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Iio [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within a (Iio a)) (nhds_within (a⁻¹) (Ioi (a⁻¹))) :=
 (continuous_inv.tendsto a).inf $ by simp [tendsto_principal_principal]
 
-@[to_additive] lemma tendsto_inv_nhds_within_Ioi_inv {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Ioi_inv [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within (a⁻¹) (Ioi (a⁻¹))) (nhds_within a (Iio a)) :=
 by simpa only [inv_inv] using @tendsto_inv_nhds_within_Ioi _ _ _ _ (a⁻¹)
 
-@[to_additive] lemma tendsto_inv_nhds_within_Iio_inv {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Iio_inv [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within (a⁻¹) (Iio (a⁻¹))) (nhds_within a (Ioi a)) :=
 by simpa only [inv_inv] using @tendsto_inv_nhds_within_Iio _ _ _ _ (a⁻¹)
+
+lemma continuous_at_iff_continuous_left_right [topological_space α] [linear_order α]
+  [topological_space β] {a : α} {f : α → β} :
+  continuous_at f a ↔ continuous_within_at f (Iic a) a ∧ continuous_within_at f (Ici a) a :=
```

Comment:
As I explained [on Zulip](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/nhds_left.20and.20nhds_right/near/205354204), I think there should be some version of
`lemma nhds_left_sup_nhds_right (x : ℝ) : 𝓝ₗ x ⊔ 𝓝ᵣ x = 𝓝 x` (with `𝓝ₗ`  and ` 𝓝ᵣ x` replaced with the relevant filters) so that the proof here becomes 
```lean
lemma tendsto_nhds_iff_left_right {α : Type*} {F : filter α} {x : ℝ} {f : ℝ → α} :
  tendsto f (𝓝 x) F ↔ tendsto f (𝓝ₗ x) F ∧ tendsto f (𝓝ᵣ x) F :=
begin
  rw ← nhds_left_sup_nhds_right,
  exact tendsto_sup
end
```
(it also means first having such a lemma in terms of `tendsto` and then deduce the `continuous_at` version which is more specific).

Reply:
Is it better now ?